lib/features/modeling/behavior/CompensateBoundaryEventBehaviour.js

```diff
@@ -43,18 +82,53 @@ export default function CompensateBoundaryEventBehaviour(eventBus, modeling) {
         oldSource = context.oldSource,
         oldTarget = context.oldTarget;
 
-    // add `isForCompensation` to new target
-    addIsForCompensationProperty(newSource, newTarget);
 
-    // remove `isForCompensation` from old target
-    removeIsForCompensationProperty(oldSource, oldTarget);
+    if (isCompensationBoundaryEvent(oldSource)) {
+
+      // remove `isForCompensation` from old target
+      removeIsForCompensationProperty(oldTarget);
+    }
+
+    if (isCompensationBoundaryEvent(newSource)) {
+
+      // add `isForCompensation` to new target
+      addIsForCompensationProperty(newTarget);
+
+      if (!replacing) {
+        removeIlegalConnections(newTarget);
+      }
+
+    }
   }, true);
 
-  this.postExecute('connection.delete', function(context) {
-    var source = context.source,
-        target = context.target;
+  this.postExecute('shape.replace', function(context) {
+    var oldShape = context.oldShape,
+        newShape = context.newShape;
+
+    if (replacing) {
+      replacing = false;
```

Comment:
What do we want to accomplish here, and why?

The way we'd usually transport such context is by passing it via `hints` to the modeling API.

Reply:
By removing the illegal connections on` connection.create`, it was causing issues when replacing. Likely because replacing involves removing the current connection and creating a new one. So with that I'm trying to pause the behaviour while replacing so it doesn't mess up the replace.

I am interested in finding a better way to do this. I tried to stick to API but ended up with this solution

Reply:
We discussed today how you could react to `a task becomes a compensate activity` to remove incoming and outgoing flows.

This way you could simplify your implementation, and get rid of the dependency tracking.

Reply:
Fyi: it isn't so simple after all. Because `a task becomes a compensate activity` after reacting to the `connection.create`, which is the problem.

Me and Maciej are looking into this together